### PR TITLE
[KOGITO-1852]: Update check for internal registry

### PR DESCRIPTION
**Pull request by radtriste**
https://issues.redhat.com/browse/KOGITO-1852

https://github.com/kiegroup/kogito-cloud-operator/pull/304 was not enough
we need also to confirm that the image pull secret is correctly attached to the pod...

Many thanks for submiting your Pull Request :heart:! 

Please make sure that your PR meets the following requirements:

- [x] You have read the [contributors guide](CONTRIBUTING.MD#sending-a-pull-request)
- [x] Pull Request title is properly formatted: `[KOGITO-XYZ] Subject`
- [x] Pull Request contains link to the JIRA issue
- [x] Pull Request contains description of the issue
- [x] Pull Request does not include fixes for issues other than the main ticket
- [ ] Your feature/bug fix has a unit test that verifies it
- [ ] You've tested the new feature/bug fix in an actual OpenShift cluster

### CODE REVIEW

**Comment by ricardozanini** [comment]
Wait. Is this a WIP?

File: test/framework/operator.go [hunk 1]
```diff
@@ -120,10 +122,47 @@ func IsKogitoOperatorRunning(namespace string) (bool, error) {
...
+			running, err := IsKogitoOperatorRunning(namespace)
+			if err != nil {
+				return false, err
+			}
+
+			// If not running, make sure the image pull secret is present in pod
+			// If not present, delete the pod to allow its reconstruction with correct pull secret
+			// Note that this is specific to Openshift
+			if !running {
+				podList, err := GetPodsWithLabels(namespace, map[string]string{"name": "kogito-operator"})
```
Comment (Sgitario) [inline test/framework/operator.go:134]: "kogito-operator" string is in Operator.Name.

File: test/framework/operator.go [hunk 2]
```diff
@@ -120,10 +122,47 @@ func IsKogitoOperatorRunning(namespace string) (bool, error) {
...
+							return false, nil
+						}
+					}
+				}
+			}
+			return running, nil
 		})
 }
 
+func podHasImagePullSecretPrefix(pod *corev1.Pod, imagePullSecretPrefix string) bool {
```
Comment (Sgitario) [inline test/framework/operator.go:155]: This method should be part of the kubernetes script instead?

**Comment by radtriste** [comment]
@ricardozanini no worry, almost done.

@Sgitario it is totally because sometimes Openshift is too slow ...
The first fix was checking that the image pull secret was created into the namespace before creating the operator. But going further in the tests, into the Kogito Openshift, we noticed that sometimes it happens that, even if the secret is created, it is not attached to the pod when operator pod is created.
The idea of the fix is really to make sure that the secret is attached. If it is not, the pod is switched into `ErrImagePull` and we get a `authentication required`, and so the scenario will fail ...

So we have 2 things now that are done:

1. We wait for the image pull secret to be created.
2. Waiting for Kogito operator, we check the secret is attached to the pod.

workaround #2 will just almost never happen. It is mostly for really low clusters...

File: test/framework/operator.go (hunk 2 above)
Comment (radtriste) [reply test/framework/operator.go:155]: will change that

**Comment by radtriste** [comment]
tested on Kogito Openshift. Took 4 hours but no ErrImagePull :)

File: test/framework/kubernetes.go [hunk 3]
```diff
@@ -218,3 +218,15 @@ func CreateSecret(namespace, name string, secretContent map[string]string) error
 
 	return kubernetes.ResourceC(kubeClient).Create(secret)
 }
+
+// CheckPodHasImagePullSecretWithPrefix checks that a pod has an image pull secret starting with the given prefix
+func CheckPodHasImagePullSecretWithPrefix(pod *corev1.Pod, imagePullSecretPrefix string) bool {
+	if len(pod.Spec.ImagePullSecrets) > 0 {
```
Comment (MarianMacik) [inline test/framework/kubernetes.go:224]: Just a note that this is unnecessary even if ImagePullSecrets is nil.

Comment (radtriste) [reply test/framework/kubernetes.go:224]: oh right ! will remove that

**Comment by spolti** [comment]
Is this PR ready?

**Pull request merged**